### PR TITLE
avoid ImageSharpen node bug on macOS

### PR DESCRIPTION
**Pull request by bigcat88**
Revert [change](https://github.com/comfyanonymous/ComfyUI/commit/4ee9aad6caf4c23e69e9e166573c505e757062e5) only for macOS

Resolves: #3383

### CODE REVIEW

**Comment by comfyanonymous** [comment]
Why not just check if the torch device is MPS?

**Comment by bigcat88** [comment]
I can’t imagine someone running Comfy on a CPU on macOS(PyTorch 2.3 dropped x86 macOS support as I know)

Anyway the question is good, I changed code to check for MPS device.

**Comment by bigcat88** [comment]
@comfyanonymous sorry to bother you, but are you fine with this PR or not?

File: comfy_extras/nodes_post_processing.py [hunk 1]
```diff
@@ -242,7 +243,7 @@ def sharpen(self, image: torch.Tensor, sharpen_radius: int, sigma:float, alpha:
 
         result = torch.clamp(sharpened, 0, 1)
 
-        return (result.to(comfy.model_management.intermediate_device()),)
+        return (result,) if comfy.model_management.mps_mode() else (result.to(comfy.model_management.intermediate_device()),)
```
Comment (shawnington) [inline comfy_extras/nodes_post_processing.py]: looking at this line makes my eyes bleed, what was wrong with a simple blocking statement for readability?

```
if comfy.model_management.mps_mode():
    return (result,)

return (result.to(comfy.model_management.intermediate_device()),)
```

Other than that the code looks like it should work as intended.

Comment (bigcat88) [reply comfy_extras/nodes_post_processing.py]: https://docs.astral.sh/ruff/rules/if-else-block-instead-of-if-exp/

Comment (bigcat88) [reply comfy_extras/nodes_post_processing.py]: changed the code style as you asked

**Pull request closed without merging**